app/javascript/components/supplemental_application/actions.js

```diff
@@ -1,32 +1,42 @@
 import apiService from '~/apiService'
 import domainToApi from '~/components/mappers/domainToApi'
 import Alerts from '~/components/Alerts'
-import { isEmpty, find, isEqual, every } from 'lodash'
+import { isEmpty, find, isEqual, every, reject } from 'lodash'
 
 export const updateApplication = async (application, prevApplication) => {
   const primaryApplicantContact = application.applicant && application.applicant.id
-  const promises = [updateLease(application['lease'], primaryApplicantContact, application['id'])]
-  // Concat lease promise with rental assistances
-  promises.concat(updateUnsavedRentalAssistances(application, prevApplication))
+
   const applicationApi = domainToApi.buildApplicationShape(application)
-  promises.push(apiService.submitApplication(applicationApi, true))
-  const responses = await Promise.all(promises)
 
-  if (every(responses, (promise) => promise !== false)) {
-    const [ { lease }, { application } ] = responses
+  const initialResponses = await Promise.all([
+    updateLease(application['lease'], primaryApplicantContact, application['id']),
+    apiService.submitApplication(applicationApi, true)
+  ])
+
+  const rentalAssistanceResponses = await Promise.all(updateUnsavedRentalAssistances(application, prevApplication))
```

Comment:
so here we're saying, first we're going to update the rental assistances, wait for a response, then update the lease and application? Due to potential collisions with creating a lease through updating a lease and saving new rental assistances without a lease already on the application and the same time?

If that's the case, can we add a comment explaining that that's why we're keeping them separate? Otherwise I think it's weird enough that it will be really confusing when we end up back in this function again.

Reply:
Currently we await the lease and application updates first, then update the rentalAssistances

This was based on our discussion where Lease and Application would take priority, I can add a comment to explain the order but I am still not familiar enough with the back end data structures and how they relate in terms of potential collisions other than what we have discussed